### ISSUE TITLE
COMMUNITY SURVEY FOR FUTURE DEVELOPMENT - ANYONE IS WELCOME TO ANSWER

### ISSUE THREAD

**Issue by lochidev** [lochidev/TextToolsPro#18]
Hey guys, for now I only have a couple of questions.
1. How long have you been using the app?
2. Does the accessibilityservice work properly on your phone? Does it get turned off automatically on your phone?

TIA and more questions to follow.

**Comment by noblereign** [comment]
just passing by and noticed this, i've been using the app since the first play store release i believe and i've been having no problems. accessibility service stays on and works properly. :+1:   
floating notes app solution sounds like an okay-ish workaround although definitely less than ideal, it'd be cool if you could figure out what those other apps do differently and try to support it, but i wouldn't be surprised if its just a system limitation or something

**Comment by lochidev** [comment]
1, 2: That's great to hear :D
3: Yeah, it's a system limitation I think. But I'll try some more when I have time. I'm trying to implement espanso forms and so I'm messing around with overlays. Maybe some kind of overlay can help? It'll still be a workaround tho similar to the floating notes app :(

**Comment by sujansujan** [comment]
1. from the initial play store release
2. yes it works. i haven't noticed my phone closing accessibility. so YES.
3. i don't quite understand it but if it helps me use this app on places where i cant use this app, then it has my support.

**Comment by lochidev** [comment]
> 1. from the initial play store release
> 
>     2. yes it works. i haven't noticed my phone closing accessibility. so YES.
> 
>     3. i don't quite understand it but if it helps me use this app on places where i cant use this app, then it has my support.

for 3: do you think a built in overlay would do any good? The solution for now is using a floating notes app (overlay from another app) like: https://f-droid.org/en/packages/genius.mohammad.floating.stickies/ 

- Personally I think a 3rd party app would do the trick for some apps. Making this built in would complicate things I guess.

**Comment by sujansujan** [comment]
> > 1. from the initial play store release
> >    
> >    2. yes it works. i haven't noticed my phone closing accessibility. so YES.
> >    3. i don't quite understand it but if it helps me use this app on places where i cant use this app, then it has my support.
> 
> for 3: do you think a built in overlay would do any good? The solution for now is using a floating notes app (overlay from another app) like: https://f-droid.org/en/packages/genius.mohammad.floating.stickies/
> 
> * Personally I think a 3rd party app would do the trick for some apps. Making this built in would complicate things I guess.

keep it as simple as you can. personally, i have been selecting apps that do work with your app for note taking. i am quite happy with what i have landed on. Its twosapp.com if you want to know.

**Comment by yephny** [comment]
1. for a few months.
2. yes.
3. n/a

I like your app, I'm so glad someone took their time to develop something like this on Android. I wish it had an official F-Droid release though.

**Comment by DeMoN100** [comment]
1. I think over 6 months now.
2. A few of my devices it will keeps shutting off periodically but then acts fine for awhile.
3. It's a workaround still need a solid solution I haven't had issues yet where I need to use this.

Addition I am noticing in some apps when you type expandroid won't work for some reason & I have to either backspace or if that does not work make a space then the expandroid will trigger.

**Comment by BlueDrink9** [comment]
1. about a year
2. Haven't noticed any issues on my lineageOS
3. Keep the app doing what it does well don't include a floating box feature

Thanks for providing the app! I like using it

**Comment by lochidev** [comment]
added another question.

**Comment by yephny** [comment]
What would be the point of 5?

**Comment by lochidev** [comment]
> What would be the point of 5?

Honestly, it is something that i would use myself as a privacy tool if I get a chance to actually talk with someone using PGP. Think PGP msges on signal. However i realize that nay I should make an entirely new app that does this without adding this feature into this app.

**Comment by yephny** [comment]
I agree, a separate app makes sense. There's also OpenKeychain on F-Droid.

**Comment by DeMoN100** [comment]
To answer the new question added no. Open keychain on f-droid can do this along with other apps available. I think it's important to fix the bugs & find solutions that fix the problems & arent temporary workarounds. Every update I still see browsers can't use it I understand why & the explanation also the temporary workaround but has any research been done into finding a permanent fix so you can do this in address bars of browsers?

Certain apps like Amethyst a Nostr client has a hiccup which I mentioned previously. That you need to make a space first then expanddroid will work. Every update I have not seen these issues fixed or any discussion about research into how to fix or if possible to use accessibility features to get around this. I think it's important to stay on the one purpose of this app & make sure it works with every app.

**Comment by DeMoN100** [comment]
> > To answer the new question added no. Open keychain on f-droid can do this along with other apps available. I think it's important to fix the bugs & find solutions that fix the problems & arent temporary workarounds. Every update I still see browsers can't use it I understand why & the explanation also the temporary workaround but has any research been done into finding a permanent fix so you can do this in address bars of browsers?
> > Certain apps like Amethyst a Nostr client has a hiccup which I mentioned previously. That you need to make a space first then expandroid will work. Every update I have not seen these issues fixed or any discussion about research into how to fix or if possible to use accessibility features to get around this. I think it's important to stay on the one purpose of this app & make sure it works with every app.
> 
> Yeah I guess I do stupid things when I get excited. Clearly not the place for question 5. I removed question 5.
> 
> To answer your question, I honestly don't have time to do any research. It simply isn't in my interest. I'm just adding features that I like making, to this app while fulfilling issues that I think are worthwhile fixing. Fixing these odd issues on browsers and youtube app search bar for example, looks like a worthwhile endeavor but tbh I'm not really interested in doing that because it looks like it will take so much time and I'm not even sure if it is possible to fix.
> 
> However, I am willing to fix these bugs if people can confirm that other text expansion apps have no issue into the specific bug they want fixed in a specific app (Please open separate issues). I would also love pull requests from others (But make sure to inform me first on what you are about to do). To the nostr hiccup, have you ensured that the espanso word property has not been set to true in your triggers? You can check that by exporting your config.

I'm sorry to read that perfecting the app is not within your interests. I understand that certain bugs are a fickle & require time to research understanding the issue & experimenting to see how it can be fixed. Anything is possible with the correct code to address it I wish I had enough knowledge so I could try to assist but it's beyond me. 

As to your question I am not using esperano expressions just the regular one & it's only when composing notes sometimes I don't have to hit space first then it will work other times I do have to. (way more do then don't) Maybe to get more open contribution is to setup a forum & make a outline of everything?

I prefer a forum since it's great for catgorization & keeping topics organized. Maybe use Zulip which is a self hostable forum that does have an mobile app & is on f-droid you can use the app to connect to your instance login & forum from your phone. I highly recommend it because of how ease the navigation is on the app you don't see forums like this nowadays. Usually something self hosted or at least on a open source app (you can just create a forum on zulip.com but their the host it's still yours but with limits because your not self hosting) will encourage more people to join because it's in a space which can be clearly recognized as independent research.

You know how some devs would never touch github & only use either forgejo, gitlabs, or gitea so I think it's always best to have research work present in a different place so more people are encouraged by seeing it's independent of any major chains. I hope you change your mind & become encouraged to face the challenges you've made a wonderful app it's almost there! I would help with advertising the project if the forum is set. You could also post some info in mines & even join the matrix & reach out by sharing your interests in [Coding Simpletons](https://rentry.org/codingsimpletons).
